bcgov/biohubbc
# Validation updates to better support UTMs and regex rules

# Overview

https://quartech.atlassian.net/browse/BHBC-1428
https://quartech.atlassian.net/browse/BHBC-1460

## This PR contains the following changes

### Validation/Transformation
- Update format validation schema to support regex flags
  - Update validation schemas to have updated rules for waypoint and utm zone
  - Update spatial-utils to allow UTM zone letters to be optional
- Update transformation/validation schemas for both moose templates
  - Recruitment should now work 
- Update XLSX transformation logic to support better support conditional rules
  - This should also fix BHBC-1460

### N8N
- Pin n8n version to 0.144.0
- Update n8n workflow exports (latest n8n version has modified the config)

### Scraping Occurrences
- Update occurrence table to add `sex` column
- Update occurrence popup to include the sex field
- Tweak occurrence popup to better handle occurrences with no geometry

## This PR contains the following types of changes

- [x] Enhancement (improvements to existing functionality)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General


### Code


### Style


### Documentation


### Tests


### Linting/Formatting

       _See the `lint` commands in package.json_
       _See the `format` commands in package.json_

### SonarCloud


## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.
